spec/awful/completion_spec.lua

```diff
@@ -15,7 +15,38 @@ if not has_bash and not has_zsh then
     return
 end
 
+local orig_path = GLib.getenv("PATH")
+local required_commands = {"bash", "zsh", "sort"}
+
+-- Change PATH
+local function get_test_path_dir()
+    local test_path = Gio.File.new_tmp("awesome-tests-path-XXXXXX")
+    test_path:delete()
+    test_path = test_path:get_path()
+    gfs.make_directories(test_path)
+
+    for _, cmd in ipairs(required_commands) do
+        local target = GLib.find_program_in_path(cmd)
+        if target then
+            os.execute(string.format('ln -s %s %s/%s', target, test_path, cmd))
+        end
```

Comment:
According to https://developer.gnome.org/glib/unstable/glib-Miscellaneous-Utility-Functions.html#g-find-program-in-path, if file is not found, NULL is returned, hence the `if target`. Since you also check in every test if bash/zsh is available, a missing file would not cause an error.

Regarding the symlink, I now triple checked the API and haven't found a `link` method or anything that resembles this functionality ... But it has a `chmod` method. Should I stick to my original plan and create executable files that call the appropriate command?

    local link = io.open(test_path .. "/" .. cmd, "w")
    link:write("exec " .. target .. ' "$@"')
    link:close()
    GLib.chmod(test_path .. "/" .. cmd, 0700)

(Not sure about the octal representation of the file mode, I would test it if you want to go with this solution.)